### ISSUE TITLE
Win 32bit - FAILED json: cannot unmarshal number 2147483648 into Go value of type int

### ISSUE THREAD

**Issue by tnakajo** [cloudfoundry/cli#1370]
## Command  

cf stop, cf logs, cf apps work but
cf restart,  cf start, cf app commands don't work
when It is Cloud Foundry CLI v6.36.1 Win 32bit.

C:¥Users¥000239>cf app APP_NAME

## What occurred

json: cannot unmarshal number 2147483648 into Go value of type int
FAILED

## CLI Version  

Cloud Foundry CLI v6.36.1 Win 32bit.
   cf version 6.36.1+e3799ad7e.2018-04-04
and 
   cf version 6.26.0+9c9a261.2017-04-06

## CC API Endpoint Version  

cf api
api endpoint: https://api.ng.bluemix.net
api version: 2.92.0

## Platform & Shell Details  

Windows 7 Professional Service Pack 1
Intel CPU/2.90GHz, 4GB memory, 32 bit operating system

**Comment by tnakajo** [comment]
Additional Information:

C:¥Users¥000239>cf -v
cf version 6.26.0+9c9a261.2017-04-06

C:¥Users¥000239>cf app APP_NAME
....
json: cannot unmarshal number 2147483648 into Go value of type int
FAILED

**Comment by tnakajo** [comment]
>cf api
api endpoint: https://api.ng.bluemix.net
api version: 2.92.0
![cli_ver_6 36](https://user-images.githubusercontent.com/23468032/39280094-b09f6450-4940-11e8-9cf0-7eefff0e09e1.png)

**Comment by tnakajo** [comment]
Windows 7 Professional Service Pack 1
Intel CPU@2.90GHz, 4GB memory, 32 bit operating system
![host_6 36](https://user-images.githubusercontent.com/23468032/39280139-ea7c319e-4940-11e8-88e4-e130d364a860.png)

**Comment by tnakajo** [comment]
[6.36_cli_trace.txt](https://github.com/cloudfoundry/cli/files/1949288/6.36_cli_trace.txt)
CLI Trace file

**Comment by XenoPhex** [comment]
@tnakajo thanks for creating this issue. We will prioritise this issue for engineering review. As an aside, we do plan in the future to drop support for 32-bit - no immediate plans to do so but wanted to mention this so that folks can prepare for that eventuality.

**Comment by tnakajo** [comment]
@XenoPhex Thank you for your comment. I am from IBM Cloud support. I posted this issue on behalf of a customer so I will share this with the customer.

**Comment by bmdu1569** [comment]
Hello @XenoPhex @tnakajo @abbyachau 
I have a similar error reported by a PAYG customer. 
They are on Windows also [I haven't yet confirmed version/architecture etc]
But they also see:
```
Uploading complete
Stopping instance 14eb0382-db44-4b8b-9961-a01998de2261
Destroying container
Successfully destroyed container

Waiting for app to start...

json: cannot unmarshal number 2147483648 into Go struct field .disk_quota of type int
FAILED 
```

Is this a cf issue?  [that is being tracked using below ticket]
https://www.pivotaltracker.com/n/projects/892938/stories/157074497

Do you know is there any workaround?

Thanks

**Comment by abbyachau** [comment]
Hi @bmdu1569 thanks for reporting - the tracker issue is correct. We've scheduled the work but have yet been able to review the issue. Are you using the same version of the CF CLI as @tnakajo?

**Comment by tnakajo** [comment]
@abbyachau See the CLI Version I reported in the description at the top.

> CLI Version
> Cloud Foundry CLI v6.36.1 Win 32bit.
> cf version 6.36.1+e3799ad7e.2018-04-04
> and
> cf version 6.26.0+9c9a261.2017-04-06

**Comment by abbyachau** [comment]
Hi @tnakajo please could you let us know if you are still running into this issue using the latest version of the CLI? Thanks.

**Comment by tnakajo** [comment]
@abbyachau I am not using Windows PC at all. It's the customer's issue via a support ticket. It was long time ago I can't confirm with them.

**Comment by abbyachau** [comment]
Thanks @tnakajo, the CLI engineering team would need more details in order to attempt to reproduce the issue. Are you able to get in touch with your customer for more details? (steps to reproduce, etc)

**Comment by AkihiroKitada** [comment]
Hello @abbyachau 

Thank you for your response.

I also could reproduce this issue on 32 bit Linux platform (CentOS 6).

**Comment by Ema93sh** [comment]
Hi @AkihiroKitada @tnakajo,

Thanks for creating this issue. The reason for failure is that the `mem_quota` field in [actor/v2action/application_instance_with_stats.go](https://github.com/cloudfoundry/cli/blob/6c08c83baf0767568dac14e0eb785e8ab4a7ff70/actor/v2action/application_instance_with_stats.go#L37) was not set to`int64` type. We pushed a [commit](https://github.com/cloudfoundry/cli/commit/2b23231b6169defc152f04b5e54265aaceac98e8#diff-d30bd7d3b400fdd52a0b01154402eeedL37) that would fix this issue. We found similar issue with other fields, which would be fixed as part of a separate story @abbyachau

**Comment by AkihiroKitada** [comment]
Hello @Ema93sh 

Thank you for your working.

Yes, actually, my customer met the same kind of issue by setting 'disk_quota' more than 2GB. I think all the other fields which type is 'int' should be 'int64' in your modified source code, in terms of make the behavior same between 32 bit platform and 64 bit platform.

**Comment by abbyachau** [comment]
Hi everyone, we've implemented and accepted a fix for this bug. We are also looking at other properties (thanks @AkihiroKitada) and are fixing those issues in a separate [story](https://www.pivotaltracker.com/story/show/163397517).  Please feel free to reach out if you have any questions/feedback.

**Comment by abbyachau** [comment]
Hi @tnakajo thanks for creating this issue. We will be releasing this fix soon in CLI `v6.43` which will also include the fix for other [properties](https://www.pivotaltracker.com/n/projects/892938/stories/163397517).  I'm closing this issue as fixed, please do not hesitate to get in touch if you have feedback. Thanks again.
